obspy/io/gcf/core.py

```diff
@@ -33,68 +142,584 @@ def merge_gcf_stream(st):
+   else:
+      # Load shared library
+      try:
+         gcf_io = _load_cdll("gcf")
+      except Exception as e:
+         print(str(e))
+      else:
+         # declare function argument and return types
+         gcf_io.read_gcf.argtypes = [ctypes.c_char_p, ctypes.POINTER(_GcfFile), ctypes.c_int]
+         gcf_io.read_gcf.restype = ctypes.c_int
+         gcf_io.free_GcfFile.argtypes = [ctypes.POINTER(_GcfFile)]
+         gcf_io.free_GcfFile.restype = None
 
+         # Decode first block
+         obj = _GcfFile()
+         b_filename = filename.encode('utf-8')
+         ret = gcf_io.read_gcf(b_filename, obj, 3)
+         if ret or (obj.n_errHead and obj.seg[0].err not in (10, 11, 21)) or obj.n_errData: 
+            is_gcf = False
+            
+         # release allocated memory
+         gcf_io.free_GcfFile(obj)
+   return is_gcf
 
-def _read_gcf(filename, headonly=False, **kwargs):  # @UnusedVariable
-    """
-    Reads a GCF file and returns a Stream object.
 
-    only GCF files containing data records are supported.
-
-    .. warning::
+def _read_gcf(filename, networkcode='', stationcode='', locationcode='', bandcode="H", instrumentcode="H", channel_prefix=None, 
+              blockmerge=True, headonly=False, cleanoverlap=True, errorret=False, **kwargs):
+   """
+   Reads a GCF file and returns a :class:`~obspy.core.stream.Stream`
+   
+   Only GCF data records are supported. Function supports format as described by 
+   GCF Reference `SWA-RFC-GCFR Issue F, December 2021 <https://www.guralp.com/apps/ok?doc=GCF_format>`
+      
+   .. warning::
         This function should NOT be called directly, it registers via the
         ObsPy :func:`~obspy.core.stream.read` function, call this instead.
-
-    :type filename: str
-    :param filename: GCF file to be read.
-    :type headonly: bool, optional
-    :param headonly: If True read only head of GCF file.
-    :type channel_prefix: str, optional
-    :param channel_prefix: Channel band and instrument codes.
-        Defaults to ``HH``.
-    :rtype: :class:`~obspy.core.stream.Stream`
-    :returns: Stream object containing header and data.
-
-    .. rubric:: Example
+   
+   :type filename: str
+   :param filename: path to GCF file to read
+   :type networkcode: str, optional
+   :param networkcode: network code to use 
+   :type stationcode: str, optional
+   :param stationcode: station code to use, if not specified unitID in gcf header will be used (first 4
+   :   characters in streamID)
+   :type locationcode: str, optional
+   :param locationcode: location code to use
+   :type bandcode: str, optional
+   :param bandcode: 1-character band code to use, ignored if channel_prefix is input
+   :type instrumentcode: str, optional
+   :param instrumentcode: 1-character instrument code to use, ignored if channel_prefix is input
+   :type channel_prefix: str, optional
+   :param channel_prefix: 2-character channel prefix to use, if not input channel_prefix = bandcode+instrumentcode
```

Comment:
I would strongly recommend to not offer clashing kwargs. Do the data ever contain band and/or instrument code? If they might do in some cases and don't in others I think we should get rid of `channel_prefix` to allow partial overrides. If band and instrument code are never recoverable from the data, i would get rid of band+instrument code kwargs and just keep `channel_prefix`

Reply:
Well personally I'd like to automatically set the channel_code according to SEED specification but that would require knowing the instrument natural period which is not contained in the GCF header. If you prefer to stick with the `channel_prefix` I can fix this. Just wondering if its not better to use the standard names to eventually be able to phase out (deprecate) the non-standard `channel_prefix` argument?

Reply:
You could consider adding a kwarg to pass in an `Inventory` to provide that information.. if you really wanna open that box. We could check how often band code and instrument code get used across the code base, but I think it might only be very few occurrences, so it might be more worthwhile to have one single kwarg instead of two for the job. But either way is fine for me. I just dont want three clashing kwargs =)

Reply:
Mmmm, passing an `Inventory` seems like using an elephant gun to shot mosquito's. A quick scan of the master branch yields no further use of `instrumentcode` nor `bandcode`, but then again neither if `channel_prefix` so presumably then, in order not break backwards compatibility with any code actually using this I guess it's the best choice is to stick with `channel_prefix`, will fix it after NGWM next week.

Reply:
done